nest-cli/src/commands/mod.rs

```diff
@@ -1,9 +1,61 @@
-use failure::Error;
+mod download;
+mod pull;
+
+pub use self::pull::pull;
+
+use colored::*;
+use failure::{Error, ResultExt};
+use std::io::{self, Write};
 
 use libnest::config::Config;
+use libnest::transaction::Transaction;
 
-pub fn pull(_config: &Config) -> Result<(), Error> {
-    Ok(())
+pub fn print_transactions(transactions: &[Transaction]) {
+    println!(
+        "{}",
+        format!(
+            "{} pending transaction{}:",
+            transactions.len(),
+            if transactions.len() <= 1 { "" } else { "s" }
+        )
+        .bold()
+    );
+    println!();
+    for transaction in transactions {
+        println!(
+            "{}",
+            match transaction {
+                Transaction::Pull(p) => {
+                    format!("{} {}", "pull".cyan(), p.target_repository().name()).bold()
+                }
+                Transaction::Install(_) => format!("{}", "install".green()).bold(),
+                Transaction::Remove(_) => format!("{}", "remove".red()).bold(),
+            }
+        );
+    }
+}
+
+pub fn ask_confirmation(question: &str, default: bool) -> Result<bool, Error> {
+    let hint = if default {
+        format!("{}/{}", "Yes".green().bold(), "no".red().bold())
+    } else {
+        format!("{}/{}", "yes".green(), "No".red())
+    };
+
+    print!("\n{} [{}] ", question.bold(), hint);
+    loop {
+        io::stdout().flush()?;
+
+        let mut input = String::new();
+        io::stdin().read_line(&mut input).context("stdin")?;
+
+        match input.trim().to_lowercase().as_ref() {
+            "" => return Ok(default),
+            "y" | "yes" | "Yes" => return Ok(true),
```

Comment:
You don't need to match for both `yes` and `Yes` because you used `to_lowercase()` above.